### PR TITLE
[rules_java] Mark v6.5.1 as compatible with `>=6.4.0`

### PR DESCRIPTION
**Pull request by fmeum**
This release requires `@bazel_tools//tools/jdk:bootstrap_runtime_toolchain_type`, which is only available as of https://github.com/bazelbuild/bazel/commit/783a50e6489f6a212ac01cfaa6f6576b51ae1c93.

### CODE REVIEW

**Comment by fmeum** [comment]
CC @hvadehra

File: modules/rules_java/6.5.1/MODULE.bazel [hunk 1]
```diff
@@ -2,6 +2,8 @@ module(
     name = "rules_java",
     version = "6.5.1",
     compatibility_level = 1,
+    # Requires @bazel_tools//tools/jdk:bootstrap_runtime_toolchain_type.
+    bazel_compatibility = [">=6.4.0"],
```
Comment (alexeagle) [inline modules/rules_java/6.5.1/MODULE.bazel:6]: Wow, that's not even released yet, seems like maybe it wasn't a wise decision in rules_java. Should we change it there instead?

**Review by alexeagle — COMMENTED** [review]
It's mutating an existing entry, so someone already using it might be broken by this? Though maybe in practice no one on <6.4.0 could have used it at all?

File: modules/rules_java/6.5.1/MODULE.bazel (hunk 1 above)
Comment (meteorcloudy) [reply modules/rules_java/6.5.1/MODULE.bazel:6]: I guess the latest rolling releases should still work with rules_java 6.5.1?

**Comment by meteorcloudy** [comment]
Ideally we should add the bazel_compatibility attribute before making the rules_java 6.5.1 release.

> It's mutating an existing entry, so someone already using it might be broken by this? Though maybe in practice no one on <6.4.0 could have used it at all?

I discussed with Fabian on slack, we think make this exception should be fine. Using Bazel older than 6.4.0 with rules_java 6.5.1 will always fail, this will make it fail with a clear error message. It won't change the final resolved graph if the resolution is successful.

**Review by meteorcloudy — APPROVED** [review]

**Comment by meteorcloudy** [comment]
The main problem here is that rules_java doesn't have a good test coverage so we didn't catch the problem when submitting 6.5.1

File: modules/rules_java/6.5.1/MODULE.bazel (hunk 1 above)
Comment (fmeum) [reply modules/rules_java/6.5.1/MODULE.bazel:6]: Yes, they do work, actually quite a few of them: https://github.com/bazelbuild/bazel/commit/8715e9ae6f5c2fd42607018b3adaad14bf601a1c

**Comment by fmeum** [comment]
@meteorcloudy Can you override the "existing module changed" check?

**Comment by meteorcloudy** [comment]
https://github.com/bazelbuild/continuous-integration/pull/1747 can help us get through presubmit

**Comment by meteorcloudy** [comment]
The test failure is expected 
```
ERROR: Bazel version 6.3.2 is not compatible with module "rules_java@6.5.1" (bazel_compatibility: [>=6.4.0])
```

**Comment by jschaf** [comment]
I just tried to add rules_java 6.5.1 via bzlmod. I saw the error about Bazel 6.4.0. But since Bazel hasn't yet released that version (delayed a week), I think I have to downgrade.

I don't think there's any action--I suspect the delayed release will resolve this next week. I'm posting in case there are any useful process nuggets here.

**Comment by meteorcloudy** [comment]
Yes, ideally we shouldn't have pushed rules_java 6.5.1 before the release of Bazel 6.4.0

**Pull request merged**